src/bitcoind.cpp

```diff
@@ -150,9 +159,20 @@ bool AppInit(int argc, char* argv[])
 
             pid_t sid = setsid();
             if (sid < 0)
-                fprintf(stderr, "Error: setsid() returned %d errno %d\n", sid, errno);
+                fprintf(stderr, "Error: setsid() failed: %s\n", strerror(errno));
+
+            (void)chdir("/");
```

Comment:
Having looked up alternative ways of dealing with absolute/relative paths, I've come up with these options:

* Remove chdir and tell daemon() not to either

* Remember PWD before changing and prepend it to all relative paths

* Abort if -daemon and any paths in the listed options are relative but include a -nochdir option to disable the chdir

* Don't chdir by default and require a -chdir or similar option to do so; should probably still abort if relative paths are provided

There are advantages to performing the chdir to / when daemonising especially if/when bitcoind becomes suitable to use as a system service so if it can be done with minimal fuss I think it should be.